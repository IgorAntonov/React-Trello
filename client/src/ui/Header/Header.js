import React from 'react';
import PropTypes from 'prop-types';

import { Row } from 'Src/ui';

export const Header = ({
  justify, bgColor, children, ...props
}) => (
  <Row
    justify={justify}
    align="center"
    padding="0.5rem"
    bgColor={bgColor}
    {...props}
  >
    {children}
  </Row>
);

Header.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.array
  ]).isRequired,
  justify: PropTypes.string.isRequired,
  bgColor: PropTypes.string.isRequired
};
