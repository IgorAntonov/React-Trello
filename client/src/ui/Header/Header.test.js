import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './Header';

describe('<Header/>', () => {
  let wrapper;
  const props = {
    bgColor: '#000',
    justify: 'center'
  };
  beforeEach(() => {
    wrapper = shallow(<Header {...props}> Header </Header>);
  });

  test('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
