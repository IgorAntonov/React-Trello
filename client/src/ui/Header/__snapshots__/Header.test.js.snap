// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match snapshot 1`] = `
<styled-layout__Row
  align="center"
  bgColor="#000"
  justify="center"
  padding="0.5rem"
>
   Header 
</styled-layout__Row>
`;
